src/Collection.php

```diff
@@ -89,17 +88,73 @@ protected function loop() {
 	/**
 	 * This function allow to format every item that is returned to the endpoint
 	 * the filter sends 3 params to the user so can be more easy to manipulate the
-	 * data based on certaim params.
+	 * data based on certain params.
 	 *
-	 * @param \WP_Post $the_post Current post object.
 	 * @return array The formated data from every item.
 	 */
-	protected function format_item( \WP_Post $the_post ) {
-		$item = [ 'id' => $the_post->ID ];
-		if ( $this->format_item ) {
-			$item = apply_filters( Filter::ITEM_FORMAT, $item, $the_post, $this->args );
+	protected function format_item() {
+		$the_post = get_post();
+
+		// Author.
+		$the_author = get_userdata( $the_post->post_author );
+
+		// Thumbnail.
+		if ( has_post_thumbnail( $the_post->ID ) ) {
+			$size = apply_filters( Filter::THUMBNAIL_SIZE, 'thumbnail', $the_post );
+
+			$attachment_id = get_post_thumbnail_id( $the_post->ID );
+
+			$src = wp_get_attachment_image_src( $attachment_id, $size );
+
+			$thumbnail = [
+				'src' 		=> $src[0],
+				'width'		=> $src[1],
+				'height'	=> $src[2],
+				'alt'		=> get_post_meta( $attachment_id, '_wp_attachment_image_alt', true ),
+			];
+		} else {
+			$thumbnail = false;
```

Comment:
Can we use the default data structure in this case: 

```
[
  'src' => '',
  'width' => '',
   'height' => ''
]
```

Or just empty array since we are using mixing values.